File: src/openprecedent/schemas/precedent.py
from pydantic import BaseModel, ConfigDict


class Precedent(BaseModel):
    model_config = ConfigDict(extra="forbid")

    case_id: str
    title: str
    summary: str
    similarity_score: int
    similarities: list[str]
    differences: list[str]
    reusable_takeaway: str | None = None
    historical_outcome: str | None = None

File: rust/openprecedent-contracts/src/precedent.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
pub struct Precedent {
    pub case_id: String,
    pub title: String,
    pub summary: String,
    pub similarity_score: i64,
    pub similarities: Vec<String>,
    pub differences: Vec<String>,
    pub reusable_takeaway: Option<String>,
    pub historical_outcome: Option<String>,
}
